bring changes from fork to master

TigraTT-Driver opened the pull request.

lannodev committed:
feat: dark theme, sfx sounds

lannodev committed:
add: preview screenshot

lannodev committed:
Update README.md

lannodev committed:
update: preview screenshot

lannodev committed:
Merge branch 'master' of https://github.com/luciano-work/neoretro

TigraTT-Driver committed:
Added some platform info

TigraTT-Driver committed:
Cleanups

TigraTT-Driver committed:
changed sound assets

TigraTT-Driver committed:
Updated Buttons
Changed onscreen-controls to vendor neutral descriptions

TigraTT-Driver committed:
Updated collection assets

TigraTT-Driver committed:
Updated theme info

TigraTT-Driver committed:
Updated Qt modules

TigraTT-Driver committed:
Translated back to English

TigraTT-Driver committed:
Changed Game-VIew Grid
Reduzed to one row to have bigger art

TigraTT-Driver committed:
Changed Home-View

Implemented composing from GameOS

TigraTT-Driver committed:
Changed  Game-View

Show in-game media for current selected game

TigraTT-Driver committed:
Updated Screenshots and Readme

TigraTT-Driver committed:
Updated Game view

Changed the composing of grid view items

TigraTT-Driver committed:
Implemented background imaages

TigraTT-Driver committed:
Not working attempt

TigraTT-Driver committed:
enhancements for smaler screens

TigraTT-Driver committed:
Merge branch 'master' into videopreview

TigraTT-Driver committed:
First working version with videopreview
for now only on games view

TigraTT-Driver committed:
Merge pull request #1 from TigraTT-Driver/videopreview

Videopreview

TigraTT-Driver committed:
Updated Readme

TigraTT-Driver committed:
Game-view selected game border fix

TigraTT-Driver committed:
Some minor changes

TigraTT-Driver committed:
Update theme.cfg

added information

TigraTT-Driver committed:
Update theme.cfg

TigraTT-Driver committed:
attempt to fix sensitive triggers

probably analoge triggers send multiple signals

TigraTT-Driver committed:
Updated sfx

TigraTT-Driver committed:
Increased icons for controls

TigraTT-Driver committed:
Analogue triggers problems

TigraTT-Driver committed:
Updated README

TigraTT-Driver committed:
Merge pull request #2 from TigraTT-Driver/testing

Minor changes

TigraTT-Driver committed:
Cleanups and layout fixes
added cleanup and fixes of @supmagc 's fork which don't change look much
some own cleanups
fixxed analouge triggers problem

TigraTT-Driver committed:
Updated Readme with new goals for settings screen

TigraTT-Driver committed:
First work for color scheme support

TigraTT-Driver committed:
more work for color scheme support

TigraTT-Driver committed:
Revert "more work for color scheme support"

This reverts commit bd503bd3910ec1f32d8aa32a43d31c5faa04349b.

TigraTT-Driver committed:
more work for color scheme support

TigraTT-Driver committed:
Updated Logos

TigraTT-Driver committed:
Save game index for returning after gaming

TigraTT-Driver committed:
Final bits for color scheme support
also added a light scheme

TigraTT-Driver committed:
Merge pull request #3 from TigraTT-Driver/testing

cleanups, fixes and color scheme support

TigraTT-Driver committed:
Update README.md

TigraTT-Driver committed:
Added settings screen
also updated README and screenshots for changes

TigraTT-Driver committed:
Merge branch 'master' into testing

TigraTT-Driver committed:
Merge pull request #4 from TigraTT-Driver/testing

Added settings screen

TigraTT-Driver committed:
Preperation for Color Logos
Fixed Settings
added color Variants of logos
ToDo: Don't corlorfill logos when colorfull is set

TigraTT-Driver committed:
Preperation for Color Logos
Fixed Settings
added color Variants of logos
ToDo: Don't corlorfill logos when colorfull is set

TigraTT-Driver committed:
Added control schemes

TigraTT-Driver committed:
Added control schemes

TigraTT-Driver committed:
Merge branch 'testing' of https://github.com/TigraTT-Driver/neoretro-dark into testing

TigraTT-Driver committed:
Merge pull request #5 from TigraTT-Driver/testing

Added on screen control schemes

TigraTT-Driver committed:
Home changes and cleanups
Integrated @supmagc fork changes for home video playback with tweaks

TigraTT-Driver committed:
Merge branch 'testing' of https://github.com/TigraTT-Driver/neoretro-dark into testing

TigraTT-Driver committed:
Merge pull request #6 from TigraTT-Driver/testing

Home changes and cleanups

TigraTT-Driver committed:
Home fixing launching games

TigraTT-Driver committed:
Merge pull request #7 from TigraTT-Driver/testing

Home fixing launching games

TigraTT-Driver committed:
more fallbacks for boxart

TigraTT-Driver committed:
Updated Settings
-Fixed defaults for logo & controll scheme
- impementen hide controls

TigraTT-Driver committed:
Merge pull request #8 from TigraTT-Driver/testing

Settings changes

TigraTT-Driver committed:
Updated color logos

TigraTT-Driver committed:
Implemented some settings
- mute sfx
- mute videos
- disable videos
- color logos switch

TigraTT-Driver committed:
collections bright colors

TigraTT-Driver committed:
Merge pull request #9 from TigraTT-Driver/testing

More settings

TigraTT-Driver committed:
All Games Collection and cleanups

TigraTT-Driver committed:
added filter for multiplayer games

TigraTT-Driver committed:
changed rating star to svg

TigraTT-Driver committed:
Added a badge for favorited games in game grid

TigraTT-Driver committed:
Cleanups

TigraTT-Driver committed:
Merge pull request #10 from TigraTT-Driver/testing

enhancements
- Added all games collection
- filter for multiplayer games
- badge on grid view for favorited games
- smaler changes

TigraTT-Driver committed:
Console colors changes
- Changed colors to reflect aspects of console color / logo color
- simplified dimm / bright handling

TigraTT-Driver committed:
Updated colorfull icons

TigraTT-Driver committed:
First Version with different language options
-ground work done
- only languages are english and german for now, need contributions

TigraTT-Driver committed:
Merge pull request #11 from TigraTT-Driver/testing

- New colorfull icons
- changed console / collection colors to bicolored
- changed console/ collection colors to reflect aspects of console color / logo color
- Implemented language support, only languages for now are english and german

TigraTT-Driver committed:
cleanup

TigraTT-Driver committed:
Reworked colorscheme support and added ozoneDark

TigraTT-Driver committed:
Added collections altcolor variant setting
other color variant is a darker form of primary collection color

TigraTT-Driver committed:
reorganized collections assets

TigraTT-Driver committed:
Merge pull request #12 from TigraTT-Driver/testing

Cleanups, reworked color scheme, collection color variation setting

TigraTT-Driver committed:
Updated logos and art

TigraTT-Driver committed:
linked alternative names for  collections

TigraTT-Driver committed:
changed language keys to be the same as on pegasus

TigraTT-Driver committed:
sorted data so missing can be added easier

TigraTT-Driver committed:
settings for future funcion
switch console assets (for now only logos) to differnet region ones.
ig switch between megadrive / genesis or different snes logos
assets are already in place, logic on menu / collection screen not

TigraTT-Driver committed:
some reorginasation

TigraTT-Driver committed:
Brought back screenshot game grid
game grid like in upstream neoretro + video as an toggle able option

TigraTT-Driver committed:
Changed all ColorOverlay ocourences
- The source property of the ColorOverlay must not point the parent item
-the source property should be made invinsible
-enabled antialiasing

TigraTT-Driver committed:
Added settings to costumize game grid size.

TigraTT-Driver committed:
Merge pull request #13 from TigraTT-Driver/testing

Merge current testing

TigraTT-Driver committed:
fix for fallback text
was incorrectly made visible for all but boxfront

TigraTT-Driver committed:
made selected item border size consistent
on all views (home, collections, games) selected item border is now 5vpx

TigraTT-Driver committed:
changes regarding logos on screenshot game grid
logos should not be visisible during videoplayback.

TigraTT-Driver committed:
Merge pull request #14 from TigraTT-Driver/testing

small changes

TigraTT-Driver committed:
upadet collection infos
- added manufacturer logos
- added some assets
- added all current plattform data

TigraTT-Driver committed:
removed symlinks

TigraTT-Driver committed:
copied back dupe collections

TigraTT-Driver committed:
Merge pull request #15 from TigraTT-Driver/testing

collections work

TigraTT-Driver committed:
update documentation

TigraTT-Driver committed:
Update HACKING.md

TigraTT-Driver committed:
Update CHANGELOG.md

TigraTT-Driver committed:
Update README.md

TigraTT-Driver committed:
Update README.md

TigraTT-Driver committed:
changes regarding background of steam games

TigraTT-Driver committed:
Updated screenshots

TigraTT-Driver committed:
Merge pull request #16 from TigraTT-Driver/testing

TigraTT-Driver committed:
Update README.md

added link to  Retro Gaming Replays theme review

TigraTT-Driver committed:
Merge pull request #17 from TigraTT-Driver/testing

Update README.md

TigraTT-Driver committed:
Collection assets region switch

TigraTT-Driver committed:
atari font was not transparent

TigraTT-Driver committed:
Added french translations, thx to @HunkDeath

fansubmaniac committed:
Add French translation

TigraTT-Driver committed:
Added support for collection videos
video file must be user provided
will be loaded from the metada defaults assets.video

TigraTT-Driver committed:
Update CHANGELOG.md

TigraTT-Driver committed:
Update theme.cfg

TigraTT-Driver committed:
Update README.md

TigraTT-Driver committed:
Merge branch 'testing' into contribution

TigraTT-Driver committed:
Merge pull request #18 from fansubmaniac/contribution

Updated French translation

TigraTT-Driver committed:
Update CHANGELOG.md

TigraTT-Driver committed:
Merge pull request #19 from TigraTT-Driver/testing

v0.133

TigraTT-Driver committed:
text for no multiplayer games if filtered

TigraTT-Driver committed:
fix for issue #20

TigraTT-Driver committed:
Cleanups and removing of sourcesize for svgs

TigraTT-Driver committed:
Merge pull request #22 from TigraTT-Driver/testing

Closes #20, smaler cleanups

TigraTT-Driver committed:
Implement #21

TigraTT-Driver committed:
Added option for dynamic number of grid rows
will set the number of rows depending on the current collection size

TigraTT-Driver committed:
Create SETTINGS.md

Some more in depth descriptions of the settings of the theme

TigraTT-Driver committed:
Update CHANGELOG.md

TigraTT-Driver committed:
Update theme.cfg

TigraTT-Driver committed:
Update README.md

TigraTT-Driver committed:
Merge pull request #23 from TigraTT-Driver/testing

Implement #21 and smaler fixes

TigraTT-Driver committed:
changed rating icons

TigraTT-Driver committed:
added seperate favorite icon

TigraTT-Driver committed:
game grid reverted default items per row to 1

TigraTT-Driver committed:
Merge pull request #24 from TigraTT-Driver/testing

small changes

TigraTT-Driver committed:
changes for upstreaming

TigraTT-Driver commented:
Hi since whe talked about this a while ago on discord, I push my changes now to your repo. If something needs to be adjusted please let me know. If you're not satisfied with the changes of the fork, i will rename it so theres less confusion for the users.

valsou commented:
Hi.

I have not reviewed all of your modifications yet. But you made an impressive and enormous work !

That being said, i decided to rework on the theme as my CS50 (Harvard MOOC) final project - branch is named reborn, i haven't pushed yet. I'm currently designing the prototype, not yet satisfied with all the positionings and features.

There will still be a "Home", "Collections" and "Games" page. Plus a "Settings" page to tweak some options.

"Games" view is a grid by default that transform to a one row ListView when Details button is pressed. Showing below the ListView few informations and assets.

I will revamp the "Collections", trying to keep it "artsy" but not relying on PathView. PathView was a headache for me...
Manufacturer is also a headache and i wonder if using a "x-manufacturer" metadata field is possible... I have some tests to do.

"Home" will be a bit different graphically, not so much in terms of functionnality.

And other ideas that were in my to-do list or that you might have already implemented.

***

I won't deliver it before March (with luck) or April, i have to re-learn some parts of QML. Some implementations will be a great challenge for me.

I don't think merging your work is the thing to do in the mean time.
You should rename it and profit from your work, if the naming confusion is an issue.

Anyways, i'd be happy to discuss theming or QML problems with you here or on Discord.

Again, great work ! Not at all disatisfied whit anything you did, your fork is better than my attempt. ;)

TigraTT-Driver commented:
Hi @valsou :smile: 

I'm not a developer and think my work is probably quite a mess. Thanks for your compliments :smile: .
I don't really want to make a profit for my work, since the basic framework comes from you. However, I don't want you to have to deal with requests from users for my chaotic work. You are also working on the theme again. Therefore, I think a renaming is best for now.

Good luck with your HOCC project. I'm looking forward to seeing what a great new concept you design for neoretro and will definitely have a look at it when it's ready from your side :wink: . If you want to adopt some of the features you wanted to do, which I have already done in my fork, but you can't find it in my chaos, just ask :+1: . If you want, maybe I can port some of my changes from the fork into the new version (when its ready :smile: ), which are lacking and I'm missing. But we'll see about that then. I won't have much time to work on themes for the next months either.

---

I really like your basic design with the three screens. It allows quick navigation and still offers a lot of information. 

- The home screen serves its purpose to get back to the game as quickly as possible. 

- The artsy Collection Screen is great.  I think the only real functional, non-cosmetic, option I added was video support. There was no Pegasus theme that had that feature yet.  But I think in many other frontends this is available and thought the users might miss it. I hope I understand your problem regarding the manufacturer, but if you want to display this information as text without a logo anyway, x-manufacturer is probably a more flexible solution for the user.

- If you take the details out of the game screen, I'll probably be one of the first to try to get them back as an option to permanently show them with the grid :grin:  What I don't like about all the other TV-like Pegasus themes is that the details are not already directly visible in the game screen. Either like you described, you have to press a button first, there is no option to show them at all or you have to navigate to an additional screen first (which I think is the worst). But that's all personal taste.  
The only thing I would change about the old concept would be the filter system. I think a screen with a selection of several filters (checkboxes favourites, multiplayer, selection of genres...) that opens at the press of a button would be best. But I haven't dared to do it yet, because I think I would have to edit the whole game screen code and I don't have the time at the moment.

- I don't know how you planned the settings screen, I just went for ListModels at the time. I think the only problem I had there afterwards was when I added language support to the theme (don't know if you also want to). I had defined the ListElements statically until then, but they didn't allow text strings from an array (from which I wanted to reload the translations). That's why the elements are now generated dynamically.
